status/reporters.go

```diff
@@ -155,6 +330,207 @@ func (r *GitHubReporter) getToken(ctx context.Context, object client.Object, nam
+	snapshot := csu.snapshot
+	scenarioName := integrationTestStatusDetail.ScenarioName
+	statusContext := NamePrefix + " / " + snapshot.Name + " / " + scenarioName
+
+	state, err := generateCommitState(integrationTestStatusDetail.Status)
+	if err != nil {
+		return nil, fmt.Errorf("unknown status %s for integrationTestScenario %s and snapshot %s/%s", integrationTestStatusDetail.Status, scenarioName, snapshot.Namespace, snapshot.Name)
+	}
+
+	description, err := generateSummary(integrationTestStatusDetail.Status, snapshot.Name, scenarioName)
+	if err != nil {
+		return nil, fmt.Errorf("unknown status %s for integrationTestScenario %s and snapshot %s/%s", integrationTestStatusDetail.Status, scenarioName, snapshot.Namespace, snapshot.Name)
+	}
+
+	return &github.CommitStatusAdapter{
+		Owner:       csu.owner,
+		Repository:  csu.repo,
+		SHA:         csu.sha,
+		State:       state,
+		Description: description,
+		Context:     statusContext,
+	}, nil
+}
+
+// updateStatusInComment will create/update a comment in PR which creates snapshot
+func (csu *CommitStatusUpdater) updateStatusInComment(ctx context.Context, integrationTestStatusDetail intgteststat.IntegrationTestStatusDetail) error {
+	issueNumberStr, found := csu.snapshot.GetAnnotations()[gitops.PipelineAsCodePullRequestAnnotation]
+	if !found {
+		return fmt.Errorf("pull-request annotation not found %q", gitops.PipelineAsCodePullRequestAnnotation)
+	}
+
+	issueNumber, err := strconv.Atoi(issueNumberStr)
+	if err != nil {
+		return err
+	}
+
+	state := integrationTestStatusDetail.Status
+	title, err := generateSummary(state, csu.snapshot.Name, integrationTestStatusDetail.ScenarioName)
+	if err != nil {
+		return fmt.Errorf(
+			"unknown status %s for integrationTestScenario %s and snapshot %s/%s",
+			integrationTestStatusDetail.Status, integrationTestStatusDetail.ScenarioName, csu.snapshot.Namespace, csu.snapshot.Name)
+	}
+
+	var comment string
+	if state == intgteststat.IntegrationTestStatusTestPassed || state == intgteststat.IntegrationTestStatusTestFail {
+		pipelineRunName := integrationTestStatusDetail.TestPipelineRunName
+		pipelineRun := &tektonv1.PipelineRun{}
+		err := csu.k8sClient.Get(ctx, types.NamespacedName{
+			Namespace: csu.snapshot.Namespace,
+			Name:      pipelineRunName,
+		}, pipelineRun)
+		if err != nil {
+			return fmt.Errorf("error while getting the pipelineRun %s: %w", pipelineRunName, err)
+		}
+
+		taskRuns, err := helpers.GetAllChildTaskRunsForPipelineRun(csu.k8sClient, ctx, pipelineRun)
+		if err != nil {
+			return fmt.Errorf("error while getting all child taskRuns from pipelineRun %s: %w", pipelineRunName, err)
+		}
+		comment, err = FormatCommentForFinishedPipelineRun(title, taskRuns)
+		if err != nil {
+			return fmt.Errorf("error while formating all child taskRuns from pipelineRun %s: %w", pipelineRun.Name, err)
+		}
+	} else {
+		comment, err = FormatCommentForDetail(title, integrationTestStatusDetail.Details)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Just an idea to further reduce complexity - the comment generation (current lines 398-431) could also be extracted to a separate function.

Reply:
step by step :D it's complicated already

Reply:
Agreed, the change does a lot already.